src/components/Player/Player.tsx

```diff
@@ -161,6 +173,10 @@ const Player: React.FC<Props> = ({
 
       playerRef.current = window.jwplayer(playerElementRef.current) as JWPlayer;
 
+      // Inject user_id and profile_id into the CDN analytics
+      const { sources, mediaid } = item;
+      attachAnalyticsParams(sources, mediaid, isJwIntegration, userId, profileId);
```

Comment:
Is there any reason you left the store lookups in this file and added them all as parameters? I think it would be safe for you to move everything into the util and just have this be a 1 line call like `attachAnalyticsParams(item)`

Reply:
Actually, the store lookups  can't be called inside of the `attachAnalyticsParams()` as it's called inside of the `useEffect` and hooks can only be called inside of the body of a function component.

Reply:
I think you can, zustand is just a little tricky and obscure about it: https://github.com/pmndrs/zustand?tab=readme-ov-file#readingwriting-state-and-reacting-to-changes-outside-of-components

You want something like `useConfigStore.getState()`.  Like you said, it won't be in the hook so it won't trigger re-renders, but integration should change and user id / profile id only change when the user logs out which probably can't happen while watching a video unless it's a background thingy triggered by inplayer sdks, in which case we probably would just want to keep logging with the user id and profile id captured at the start.

Reply:
I'll try it, thnx!